File: medium/ReverseWordsInString.cs
namespace Medium;

/// <summary>
///
/// </summary>
public class ReverseWordsInString
{
    /// <summary>
    ///
    /// </summary>
    public static string Solve(string s)
    {
        string[] words = s.Trim().Split(' ', StringSplitOptions.RemoveEmptyEntries);
        List<string> result = [];
        for (int i = words.Length - 1; i >= 0; i--)
        {
            if (words[i] != " ")
            {
                result.Add(words[i]);
            }
        }
        return string.Join(" ", result);
    }
}

File: medium/reverse_words_in_string.go
package medium

import "strings"

func ReverseWordsInString(s string) string {
	words := strings.Fields(strings.TrimSpace(s))
	var result []string
	reverse := len(words) - 1
	for reverse >= 0 {
		if words[reverse] != " " {
			result = append(result, words[reverse])
		}
		reverse--
	}

	return strings.Join(result, " ")
}
